packages/flutter/lib/src/material/material.dart

```diff
@@ -130,6 +132,12 @@ abstract class MaterialInkController {
 ///      rounded edges. The edge radii is specified by [kMaterialEdges].
 ///    - [MaterialType.transparency]: the default material shape is a rectangle.
 ///
+/// ## Border
+///
+/// If [shape] is specified, it will be used it to paint a border.
+/// [ShapeBorder] implementations like [CircleBorder] only paint a border when
+/// [CircleBorder.side] is not [BorderSide.none].
```

Comment:
I'm not sure that last sentence is useful. I'd probably phrase this as just "If [shape] is not null, then its border will also be painted (if any)."

Reply:
(then one day maybe we can add sample code or a screen shot showing this, or something)

Reply:
done